src/main.rs

```diff
@@ -163,7 +181,56 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
                 ))?;
             println!("{:#X?}", &found_struct.defined_struct())
         }
-        (None, None, None, None, true) => {
+        (None, None, None, None, true, false) => {
+            for undefined_struct in smbios_data {
+                /*
+                    Handle 0x0000, DMI type 0, 20 bytes
+                        Header and Data:
+                                00 14 00 00 01 02 00 F0 03 03 90 DA CB 7F 00 00
+                                00 00 34 01
+                        Strings:
+                                41 6D 65 72 69 63 61 6E 20 4D 65 67 61 74 72 65
+                                6E 64 73 20 49 6E 63 2E 00
+                                "American Megatrends Inc."
+                                30 39 30 30 30 38 20 00
+                                "090008 "
+                                31 32 2F 30 37 2F 32 30 31 38 00
+                                "12/07/2018"
+                */
+                println!(
+                    "Handle {:#06X}, DMI type {}, {} bytes",
+                    *undefined_struct.header.handle(),
+                    undefined_struct.header.struct_type(),
+                    undefined_struct.fields.len()
+                );
+                print!("\tHeader and Data:");
+                for item in undefined_struct.fields.iter().enumerate() {
+                    if item.0 % 16 == 0 {
+                        println!();
+                        print!("\t\t");
+                    }
+                    print!("{:02X} ", item.1);
+                }
+                println!();
+                print!("\tStrings:");
+                for string_item in undefined_struct.strings.iter() {
+                    for item in string_item.iter().enumerate() {
+                        if item.0 % 16 == 0 {
+                            println!();
+                            print!("\t\t");
+                        }
+                        print!("{:02X} ", item.1);
+                    }
+                    println!();
+                    let as_string: String = string_item.iter()
```

Comment:
@jrgerber , not sure if using String::from_utf8 might help here? Other than this comment, lgtm, Thanks